docs/content/patterns/ask-users-for-dates-and-times/index.mdx

```diff
@@ -0,0 +1,216 @@
+---
+title: Ask users for dates and times
+description: When we ask for date, time or a combination of both there are a range of components that can be used to meet user needs in different situations.
+relatedComponents: ['time-input', 'time-picker']
+---
+
+## Time input
+
+[Time input](/components/time-input) allows users to type an exact time - for example, the known time of an incident or event.
+
+When the user enters a time, formatting is applied to collect input in a consistent format.
+
+It should be used in situations when user know and can provide an exact and arbitrary time.
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState();
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	const invalid = value && !isValidTime(value.value);
+
+	return (
+		<TimeInput
+			hideOptionalLabel
+			invalid={invalid}
+			label="Time"
+			message="Enter a valid time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+## Time picker
+
+[Time picker](/components/time-picker) is a composition of [combobox](/components/combobox) that allows users to select a time from a predefined list of consistent time intervals.
+
+As the user enters text into the field, autocomplete narrows the list to match their input.
+
+The list is closed when a time is selected.
+
+It should be used when a user needs to select from a large list of specific times.
+
+```jsx live
+() => {
+	const [value, setValue] = React.useState(null);
+
+	const onChange = (timeValue) => {
+		console.log('onChange', timeValue);
+		setValue(timeValue);
+	};
+
+	return (
+		<TimePicker
+			hideOptionalLabel
+			hint="Start typing to see results"
+			label="Select a time"
+			onChange={onChange}
+			value={value}
+		/>
+	);
+};
+```
+
+## Asking for specific known dates
+
+When a specific known date is required use a format tolerant [Text input](/components/text-input) that validates acceptable formats and auto formats minor issues like spaces or capital letters.
+
+Examples of a specific known date are birth date, when an application was made or an expiry date.
+
+```jsx live
+<TextInput
+	hideOptionalLabel
+	hint="(e.g. 15/02/1982)"
+	label="Date of event"
+	type="date"
```

Comment:
We shouldn't be using `type="date"` right? We don't want the browser's native calendar